tests/drivers/tee/tee/src/main.c

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2023 Epam Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @addtogroup t_tee_driver
+ * @{
+ * @defgroup t_tee_driver test_tee_driver
+ * @}
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/ztest.h>
+#include <zephyr/device.h>
+
+#include <zephyr/drivers/tee.h>
+
+ZTEST(tee_test_suite, test_basic_calls)
+{
+	int ret;
+	uint32_t session_id;
+	int addr;
+	struct tee_shm *shm = NULL;
+	const struct device *const dev = DEVICE_DT_GET_ONE(test_tee);
+
+	zassert_not_null(dev, "Unable to get dev");
+
+	ret = tee_get_version(dev, NULL);
```

Comment:
So what exactly is tested? That you can call a function?

Reply:
Yes. That it generates syscall correctly.

Reply:
Lots of zephyr unitests check that syscalls are working.

I don't like this tests because they contradict GP TEE API specifications. Basically, for anyone reading this, it looks like you are testing to ensure that your TEE API is not compatible with GP spec. 

Remember that unit tests often are used as code examples. So code under tests should always behave to spec.

Reply:
It may sound funny, but those tests were used as code example. It shows how dummy optee driver can be implemented. This helped a lot for the team while developing optee_client. I can drop those tests is you think they are useless.